Packs/GitLab/Integrations/GitLabEventCollector/GitLabEventCollector.py

```diff
@@ -0,0 +1,129 @@
+        projects.sort(key=lambda k: k.get('id'))
+        user_events = [event for event in events if 'entity_type' not in event]
+        user_events.sort(key=lambda k: k.get('id'))
+        return {'groups': groups[-1]['created_at'], 'projects': projects[-1]['created_at'],
+                'events': user_events[-1]['created_at']}
+
+    def _iter_events(self):  # pragma: no cover
+        self.client.set_request_filter(None)
+        response = self.call()
+        events: list = response.json()
+        events.sort(key=lambda k: k.get('created_at'))
+        if not events:
+            return []
+
+        while True:
+            yield events
+            self.client.set_request_filter(None)
+            response = self.call()
+            events = response.json()
+            events.sort(key=lambda k: k.get('created_at'))
+            if not events:
+                break
+
+
+def main() -> None:  # pragma: no cover
+    demisto_params = demisto.params()
+    url = urljoin(demisto_params['url'], '/api/v4/')
+    events_collection_management = {
+        'groups_ids': argToList(demisto_params.get('group_ids', '')),
+        'projects_ids': argToList(demisto_params.get('project_ids', '')),
+        'event_types': ['groups', 'projects']
+    }
+    user_ids = demisto_params.get('user_ids', '').split(',')
+    headers = {'PRIVATE-TOKEN': demisto_params.get('api_key', {}).get('credentials', {}).get('password')}
+    request_object = {
+        'method': Method.GET,
+        'url': url,
+        'headers': headers,
+    }
+    last_run = demisto.getLastRun()
+    if ('groups', 'projects', 'events') not in last_run:
+        last_run = dateparser.parse(demisto_params['after'].strip()).isoformat()
+        last_run = {
+            'groups': last_run,
+            'projects': last_run,
+            'events': last_run,
+        }
+    else:
+        last_run = last_run
+
+    options = IntegrationOptions(**demisto_params)
+
+    request = IntegrationHTTPRequest(**request_object)
+
+    client = Client(request, options, last_run)
+
+    get_events = GetEvents(client, options)
+
+    command = demisto.command()
+    try:
+        events = []
+        for event_type in events_collection_management['event_types']:
+            for obj_id in events_collection_management[f'{event_type}_ids']:
+                call_url_suffix = f'{event_type}/{obj_id}/audit_events'
+                get_events.client.request.url = url + call_url_suffix
+                get_events.client.page = 1
+                get_events.client.event_type = event_type
+                events.extend(get_events.run())
+        get_events.client.event_type = 'events'
+        for obj_id in user_ids:
+            get_events.client.request.url = url + f'users/{obj_id}/events'
+            get_events.client.page = 1
+            events.extend(get_events.run())
+        if command == 'test-module':
+            return_results('ok')
```

Comment:
the `test-module` doesn't test anything.

Reply:
It creates the events before so if it got there it means that it worked